libjwt-typed: unbreak with minor changes

someodd opened the pull request:
Unbreak libjwt-typed. I was able to build using:

  nix-build --no-out-link -A haskellPackages.libjwt-typed .

This was accomplished by removing dependency
bounds, skipping tests, and adding the libjwt
librarySystemDepends.

Removed `broken` status from libjwt-typed.

This has not been tested in any concrete way (I am
essentially new to Nix). I would have liked to try
using these changes with a project which spurred
these changes, but I am unsure how to override in
a manner that used the local nixpkgs repo (where I
worked on these changes).

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

maralorn commented:
Thx!

2 small things.

a) you need to move the libjwt dependency from the hackage-packages.nix (which is autogenerated and will be overriden) to configuration-nix.nix. See the miniball override in that file as an example for how to do that.
b) Can you open an issue upstream informing them which bounds are out of date and put that in a comment?

someodd commented:
@maralorn thank you for your encouragement on Mastodon and Matrix. Your nice help has made this relatively painless.

Here is the issue you requested I make, in order to alert the author of `libjwt-typed` that the version bounds are out-of-date for said project: https://github.com/marcin-rzeznicki/libjwt-typed/issues/2

maralorn commented:
Nearly there. Can you put the link into a comment above the line with the override? I personally also like to add a timestamp

```
# 2022-08-07: Bounds are to restrictive: https://github.com/marcin-rzeznicki/libjwt-typed/issues/2
```

and then please rebase everything into one commit.

someodd committed:
libjwt-typed: unbreak with minor changes

Unbreak libjwt-typed. Succesfully built using:

  env NIXPKGS_ALLOW_BROKEN=1 nix-build --no-out-link -A haskellPackages.libjwt-typed

This was accomplished by removing dependency
bounds, skipping tests, and adding the libjwt
librarySystemDepends.

Removed `broken` status from libjwt-typed.

I have not tested using what was built in the
nix-build command above in a local project.

someodd commented:
Okay, it's all in one commit now, plus there's also the timestamped comment you requested, @maralorn.

maralorn commented:
Sorry to bother you again, but that comment belongs together with the jailbreak in the configuration-common.nix. After all it’s about why you do the jailbreak. The library override already had a satisfying comment.

someodd commented:
That makes sense, sorry about that, @maralorn. I have made the amended commit changing the comments. Thank you.

maralorn commented:
Thank, you!

Next time you can also run `maintainers/scripts/haskell/regenerate-hackage-packages.sh` to remove the broken flag from the hackage-packages.nix file, but we will run it later on the branch this time …
